File: Sorting/bubble_sort/Java/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
  public static void main(String[] args) {
    int[] data = { 1, 9, 6, 4, 10, 8, 2, 5, 7, 3 };
    System.out.println("Before Sort: " + Arrays.toString(data));
    data = sort(data);
    System.out.println("After Sort:  " + Arrays.toString(data));
  }

  // bubble sort algorithm
  public static int[] sort(int[] data) {
    for (int i = 0; i < data.length; i++) {
      for (int j = 0; j < data.length - 1; j++) {
        if (data[j] > data[j + 1]) {
          int temp    = data[j];
          data[j]     = data[j + 1];
          data[j + 1] = temp;
        }
      }
    }

    return data;
  }
}

File: Sorting/bubble_sort/PHP/bubblesort.php
<?php

function bubble_sort($array) {
	for ($x = 0;$x < count($array);$x++) {
		for ($y = 0;$y < count($array) - $x - 1;$y++) {
			if ($array[$y] > $array[$y + 1]) {
				$temp = $array[$y];
				$array[$y] = $array[$y + 1];
				$array[$y + 1] = $temp;
	}
}
	echo ("After #" . $x . " run:   ");
	print_array($array);
	}
}
function print_array($array) {
	for ($x = 0;$x < count($array);$x++) {
		echo ($array[$x] . " ");
	}
	echo ("\n");
}

$array = [34, 22, 1, 8, 2, 13];
echo "Original Array: ";
print_array($array);
bubble_sort($array);

?>